http-server-netty/src/main/java/io/micronaut/http/server/netty/RoutingInBoundHandler.java

```diff
@@ -386,13 +390,28 @@ private void exceptionCaughtInternal(ChannelHandlerContext ctx,
                         Object result = handler.handle(nettyHttpRequest, cause);
                         return errorResultToResponse(result);
                     });
-                    Supplier<Flowable<? extends MutableHttpResponse<?>>> publisherSupplier = () -> routePublisher;
```

Comment:
We need to be careful here not to create regressions for https://github.com/micronaut-projects/micronaut-core/commit/be1bc41b4ac037dab48a10eeb25ec6df6059fc2f

However if this tests pass probably ok

Reply:
Refactoring surprisingly eliminated the need for deferred publishers.

Reply:
@dstepanov @graemerocher This might still have caused some regression as per my investigation of a bug appeared in v2.5.1+

The current request in `ServerRequestContext` was no longer set properly in filters. Now as I see this has been fixed for HTTP requests in v2.5.2 but not for WebSocket requests, I still see this when upgrading our projects through @RequestScoped beans throwing exceptions in filters.

I just managed to track down the regression to this change, I will try to put together some sort of test to reproduce the issue tomorrow to raise the issue, but in case you would have any hints on where this distinction between HTTP and WebSocket requests is located in the codebase or point me to the right direction on how to put together a websocket based test, that would be helpful.

Reply:
@lgathy This is where filters are executed for websocket upgrade requests https://github.com/micronaut-projects/micronaut-core/blob/2.5.x/http-server-netty/src/main/java/io/micronaut/http/server/netty/websocket/NettyServerWebSocketUpgradeHandler.java#L150

Reply:
@jameskleeh Thanks for the help. I found a way to reproduce this behaviour in tests (might not be the nicest way), so I raised an issue #5900 and soon will submit a draft PR to demonstrate it.